batch/manage-pool/manage-pool-linux.sh

```diff
@@ -1,32 +1,71 @@
 az batch account login \
-    --resource-group myResourceGroup \
-    --name mybatchaccount
+    --resource-group $resourceGroup \
+    --name $batchAccount \
     --shared-key-auth
 
 # Retrieve a list of available images and node agent SKUs.
-az batch pool node-agent-skus list
+# az batch pool node-agent-skus list # fails
+az batch pool supported-images list
+#az batch pool supported-images list --query [*].imageReference.[offer,publisher]
+#az batch pool supported-images list --query [].imageReference.[offer,publisher]
+
+#az batch pool supported-images list --query [].osType
+#az batch pool supported-images list --query '[ostype, imageReference.offer[0],imageReference.publisher[0]'
+#az batch pool supported-images list --query '[imageReference.offer[0],imageReference.publisher[0]'
+
+#az batch pool supported-images list --query "[].{OS:osType,Image:imageReference.offer, Publisher:publisher, SKU:sku}"
+#az batch pool supported-images list --query "[].{OS:osType,Image:imageReference.offer, Publisher:publisher, SKU:sku}" -o table # only 2 columns
+
+# https://docs.microsoft.com/en-us/cli/azure/query-azure-cli#filter-arrays
+#az batch pool supported-images list --query "[?osType=='linux'].{OS:osType,Image:imageReference.offer, Publisher:publisher, SKU:sku}"
+# https://docs.microsoft.com/en-us/cli/azure/query-azure-cli#filter-arrays
+# az batch pool supported-images list --query "[? contains(imageReference.offer,'Ubunto')].[?osType=='linux'].{OS:osType,Image:imageReference.offer, Publisher:publisher, SKU:sku}"
+#az batch pool supported-images list --query "[? contains(imageReference.offer,"ubunto")].{OS:osType,Image:imageReference.offer, Publisher:publisher, SKU:sku}"
+#            az vm list -g QueryDemo --query "[? contains(storageProfile.osDisk.managedDisk.storageAccountType,'SSD')].{Name:name, Storage:storageProfile.osDisk.managedDisk.storageAccountType}" -o json
+
+#az batch pool supported-images list --query [].osType,[*].imageReference.[offer,publisher]
+
+#az batch pool supported-images list --query "[? contains(nodeAgentSkuId)].{batch.node.ubuntu 20.04}"
+
+#az batch pool supported-images list --query [].nodeAgentSkuId # limit to ubunto and add canonical
+#az batch pool supported-images list --query [].canonical
 
 # Create a new Linux pool with a virtual machine configuration. The image reference 
 # and node agent SKUs ID can be selected from the ouptputs of the above list command.
 # The image reference is in the format: {publisher}:{offer}:{sku}:{version} where {version} is
-# optional and defaults to 'latest'.
+# optional and defaults to 'latest'."
+
 az batch pool create \
     --id mypool-linux \
     --vm-size Standard_A1 \
-    --image canonical:ubuntuserver:16.04-LTS \
-    --node-agent-sku-id "batch.node.ubuntu 16.04"
+    --image canonical:ubuntuserver:18.04-lts \
+    --node-agent-sku-id "batch.node.ubuntu 18.04" # what is query to get correct update to --image and --node-agent-ski-id?
```

Comment:
There is not a way to directly obtain the latest image/node-agent-sku-id. You can use the query parameter to at least filter on ubuntu support images i.e:
az batch pool supported-images list --query "[?contains(imageReference.offer,'ubuntuserver') && imageReference.publisher == 'canonical'].{Offer:imageReference.offer, Publisher:imageReference.publisher, Sku:imageReference.sku, nodeAgentSkuId:nodeAgentSkuId}[-1]" --output tsv

This should return the offer, publisher, and sku of the image along with the node agent sku id as a tsv and you can utilize further Bash processing to extract these values as variables and pass them to the AZ Batch CLI command.